src/backend/ManagementHub.Service/Areas/Tournaments/TournamentsController.cs

```diff
@@ -0,0 +1,213 @@
+		ITournamentContextProvider tournamentContextProvider,
+		IUpdateTournamentBannerCommand updateTournamentBannerCommand)
+	{
+		this.contextAccessor = contextAccessor;
+		this.tournamentContextProvider = tournamentContextProvider;
+		this.updateTournamentBannerCommand = updateTournamentBannerCommand;
+	}
+
+	/// <summary>
+	/// List tournaments in the Hub.
+	/// Private tournament filtering and IsCurrentUserInvolved computation is done at the database level via joins.
+	/// </summary>
+	[HttpGet]
+	[Tags("Tournament")]
+	public async Task<Filtered<TournamentViewModel>> GetTournaments([FromQuery] FilteringParameters filtering)
+	{
+		var userContext = await this.contextAccessor.GetCurrentUserContextAsync();
+
+		// QueryTournaments performs all filtering and computation at the database level via joins:
+		// - Filters private tournaments (only shows those the user manages)
+		// - Computes IsCurrentUserInvolved based on tournament manager status
+		// Phase 3 will extend: also check if user is a participant via team manager role
+		// Phase 4 will extend: also check if user is on a roster
+		var query = this.tournamentContextProvider.QueryTournaments(userContext.UserId);
+
+		var tournaments = query.AsFiltered();
+
+		// Fetch banner URLs for all tournaments
+		var tournamentIds = tournaments.Items.Select(t => t.Id).ToList();
+		var bannerUrls = new Dictionary<TournamentIdentifier, Uri?>();
+		foreach (var tournamentId in tournamentIds)
+		{
+			var bannerUri = await this.tournamentContextProvider
+				.GetTournamentBannerUriAsync(tournamentId, this.HttpContext.RequestAborted);
+			bannerUrls[tournamentId] = bannerUri;
+		}
+
+		// Map to view models - IsCurrentUserInvolved is already computed at DB level
+		var viewModels = tournaments.Items.Select(t => new TournamentViewModel
+		{
+			Id = t.Id,
+			Name = t.Name,
+			Description = t.Description,
+			StartDate = t.StartDate,
+			EndDate = t.EndDate,
+			Type = t.Type,
+			Country = t.Country,
+			City = t.City,
+			Place = t.Place,
+			Organizer = t.Organizer,
+			IsPrivate = t.IsPrivate,
+			BannerImageUrl = bannerUrls.TryGetValue(t.Id, out var uri) ? uri?.ToString() : null,
+			IsCurrentUserInvolved = t.IsCurrentUserInvolved
+		});
+
+		return viewModels.AsFiltered();
+	}
+
+	/// <summary>
+	/// Get tournament details.
+	/// </summary>
+	[HttpGet("{tournamentId}")]
+	[Tags("Tournament")]
+	public async Task<ActionResult<TournamentViewModel>> GetTournament([FromRoute] TournamentIdentifier tournamentId)
+	{
+		var userContext = await this.contextAccessor.GetCurrentUserContextAsync();
+		var tournament = await this.tournamentContextProvider
+			.GetTournamentContextAsync(tournamentId, userContext.UserId, this.HttpContext.RequestAborted);
+
+		// Check access to private tournament - this is enforced at database level
+		// GetTournamentContextAsync already filters private tournaments based on IsCurrentUserInvolved
+		if (tournament.IsPrivate && !tournament.IsCurrentUserInvolved)
```

Comment:
This is check is redundant with the database filter right? We can keep it just in case, but change the behavior to throw the not found exception with tournament id same as it's done in the DB layer.